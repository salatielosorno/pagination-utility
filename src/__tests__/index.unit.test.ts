import { PaginationUtility } from '../index';

describe('Pagination Utility', () => {
  test('Get total pages with total - total items = 31', async () => {
    const totalItems = 31;
    const perPage = 10;

    const totalPages = PaginationUtility.getTotalPages({ totalItems, perPage });

    expect(totalPages).toMatchSnapshot();
  });

  test('Get total pages with total - total items = 30', async () => {
    const totalItems = 30;
    const perPage = 10;

    const totalPages = PaginationUtility.getTotalPages({ totalItems, perPage });

    expect(totalPages).toMatchSnapshot();
  });

  test('Get values to paginate = currentPage = 1', async () => {
    const currentPage = 1;
    const perPage = 10;

    const values = PaginationUtility.getValuesToPaginate({
      currentPage,
      perPage
    });

    expect(values).toMatchSnapshot();
  });

  test('Get values to paginate - currentPage = 5', async () => {
    const currentPage = 5;
    const perPage = 10;

    const values = PaginationUtility.getValuesToPaginate({
      currentPage,
      perPage
    });

    expect(values).toMatchSnapshot();
  });
});
