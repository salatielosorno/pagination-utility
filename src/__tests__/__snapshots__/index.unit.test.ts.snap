// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Utility Get total pages with total - total items = 30 1`] = `3`;

exports[`Pagination Utility Get total pages with total - total items = 31 1`] = `4`;

exports[`Pagination Utility Get values to paginate = currentPage = 1 1`] = `
Object {
  "limit": 10,
  "offset": 0,
}
`;

exports[`Pagination Utility Get values to paginate - currentPage = 5 1`] = `
Object {
  "limit": 10,
  "offset": 40,
}
`;
